pkg/indexer/indexer.go

```diff
@@ -0,0 +1,40 @@
+package indexer
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+type Indexer interface {
+	Object() client.Object
+	Name() string
+	Keys(client.Object) []string
+}
+
+// RegisterAll registers all known indexers to the given manager.
+// It uses the given logger to create a new named "indexer" logger,
+// passing that to each indexer.
+func RegisterAll(ctx context.Context, mgr manager.Manager, logger *zap.Logger) error {
```

Comment:
I wonder if this is the most ergonomic way. I mean we will need to remember to add the indexers here when we create new ones. Maybe it is not a problem.

Reply:
not sure either 🤷 do you have any other suggestions?

Reply:
Was thinking about self registration, using inits, but it might be worse. Let's go for this for now.

Reply:
`init` is considered an anti-pattern generally. classic dependency-injection, as done here seems a better alternative.

Reply:
i.e. `init` can have nasty side-effects, potentially invoked twice if imported under different packages.